File: standard-udfs-spark/src/main/scala/com/linkedin/stdudfs/spark/examples/StructCreateByIndexFunctionWrapper.scala
package com.linkedin.stdudfs.spark.examples

import java.util

import com.google.common.collect.ImmutableList
import com.linkedin.stdudfs.api.udf.{StdUDF, TopLevelStdUDF}
import com.linkedin.stdudfs.examples.StructCreateByIndexFunction
import com.linkedin.stdudfs.spark.StdUdfWrapper
import org.apache.spark.sql.catalyst.expressions.Expression


case class StructCreateByIndexFunctionWrapper(expressions: Seq[Expression]) extends StdUdfWrapper(expressions) {

  override protected def getTopLevelUdfClass: Class[_ <: TopLevelStdUDF] = classOf[StructCreateByIndexFunction]

  override protected def getStdUdfImplementations: util.List[_ <: StdUDF] = ImmutableList.of(
    new StructCreateByIndexFunction()
  )
}

File: standard-udfs-presto/src/main/java/com/linkedin/stdudfs/presto/examples/StructCreateByIndexFunctionWrapper.java
package com.linkedin.stdudfs.presto.examples;

import com.linkedin.stdudfs.api.udf.StdUDF;
import com.linkedin.stdudfs.examples.StructCreateByIndexFunction;
import com.linkedin.stdudfs.presto.StdUdfWrapper;


public class StructCreateByIndexFunctionWrapper extends StdUdfWrapper {
  public StructCreateByIndexFunctionWrapper() {
    super(new StructCreateByIndexFunction());
  }

  @Override
  protected StdUDF getStdUDF() {
    return new StructCreateByIndexFunction();
  }
}
